api/submissions/upload.go

```diff
@@ -46,6 +50,15 @@ func findChallenge(cs model.Challenges, id string) (model.Challenge, error) {
 	return cs.Find(cid)
 }
 
+func verifyUser(err error, r *http.Request, us model.Users, u **model.User) error {
```

Comment:
double pointer here looks weird to me - `u` variable is already a pointer, so just returning a point to User struct should be enough